src/Service/KpiSetupService.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\KpiSetup;
+use Doctrine\ORM\EntityManagerInterface;
+
+class KpiSetupService
+{
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    public function list(): array
+    {
+        try {
+            $kpiSetups = $this->entityManager->getRepository(KpiSetup::class)->findAll();
+            $data = [];
+            foreach ($kpiSetups as $kpiSetup) {
+                $data[] = [
+                    'id' => $kpiSetup->getId(),
+                    'title' => $kpiSetup->getTitle(),
+                    'status' => $kpiSetup->getStatus(),
+                ];
+            }
```

Comment:
Please use a custom query & prepare this data directly from repository instead of for loop.

Reply:
Done